web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java

```diff
@@ -181,62 +183,53 @@ public static boolean storeLocale(String locale) {
 				
 				// first we are saving locale as administrative user's property
 				if (userId != null) {
-					String insert = "insert into user_property (user_id, property, property_value) values (?, 'defaultLocale', ?)";
-					PreparedStatement statement = null;
```

Comment:
Do you mind explaining to me why you replaced the sql insert with both an update and insert?

Reply:
I replaced the sql insert with both an update and an insert to avoid primary key constraint violations when a locale already exists for the user. The update handles existing entries, while the insert adds a new one if none is found.

Reply:
A locale exists for a new database setup?

Reply:
fixes this https://pastebin.com/STThpW6y issue reported by you

Reply:
But we never had this problem before your changes.